src/inc/til/env.h

```diff
@@ -0,0 +1,575 @@
+        //      The sort is case-insensitive, Unicode order, without regard to locale.
+        //      Because the equal sign is a separator, it must not be used in the name of
+        //      an environment variable."
+        //      https://docs.microsoft.com/en-us/windows/desktop/ProcThread/changing-environment-variables
+        //
+        // - Returns CSTR_LESS_THAN, CSTR_EQUAL or CSTR_GREATER_THAN
+        [[nodiscard]] int compare_string_ordinal(const std::wstring& lhs, const std::wstring& rhs) noexcept
+        {
+            int result = CompareStringOrdinal(lhs.c_str(), -1, rhs.c_str(), -1, 1);
+            FAIL_FAST_LAST_ERROR_IF(!result);
+            return result;
+        }
+
+        struct wstring_case_insensitive_compare
+        {
+            [[nodiscard]] bool operator()(const std::wstring& lhs, const std::wstring& rhs) const noexcept
+            {
+                return compare_string_ordinal(lhs, rhs) == CSTR_LESS_THAN;
+            }
+        };
+
+        namespace vars
+        {
+            static constexpr std::wstring_view system_root{ L"SystemRoot" };
+            static constexpr std::wstring_view system_drive{ L"SystemDrive" };
+            static constexpr std::wstring_view all_users_profile{ L"ALLUSERSPROFILE" };
+            static constexpr std::wstring_view public_var{ L"PUBLIC" };
+            static constexpr std::wstring_view program_data{ L"ProgramData" };
+            static constexpr std::wstring_view computer_name{ L"COMPUTERNAME" };
+            static constexpr std::wstring_view user_name{ L"USERNAME" };
+            static constexpr std::wstring_view user_domain{ L"USERDOMAIN" };
+            static constexpr std::wstring_view user_dns_domain{ L"USERDNSDOMAIN" };
+            static constexpr std::wstring_view home_drive{ L"HOMEDRIVE" };
+            static constexpr std::wstring_view home_share{ L"HOMESHARE" };
+            static constexpr std::wstring_view home_path{ L"HOMEPATH" };
+            static constexpr std::wstring_view user_profile{ L"USERPROFILE" };
+            static constexpr std::wstring_view app_data{ L"APPDATA" };
+            static constexpr std::wstring_view local_app_data{ L"LOCALAPPDATA" };
+
+            static constexpr std::wstring_view program_files{ L"ProgramFiles" };
+            static constexpr std::wstring_view program_files_x86{ L"ProgramFiles(x86)" };
+            static constexpr std::wstring_view program_files_arm64{ L"ProgramFiles(Arm)" };
+            static constexpr std::wstring_view program_w6432{ L"ProgramW6432" };
+            static constexpr std::wstring_view common_program_files{ L"CommonProgramFiles" };
+            static constexpr std::wstring_view common_program_files_x86{ L"CommonProgramFiles(x86)" };
+            static constexpr std::wstring_view common_program_files_arm64{ L"CommonProgramFiles(Arm)" };
+            static constexpr std::wstring_view common_program_w6432{ L"CommonProgramW6432" };

```

Comment:
In headers you need to use `inline constexpr` instead of `static constexpr`, since "static" means "local to the current compilation unit", whereas "inline" means "deduplicate redundant instances of this". In other words, "static" will create copies of these constants in every file that includes this header, whereas "inline" will explicitly deduplicate them. The compiler/linker would probably deduplicate these anyways, but better safe than sorry! `inline constexpr` is a new feature in C++17.